### PR TITLE
Allow position of the cell to be passed in arguments

### PR DESCRIPTION
**Pull request by rmdort**
Cell position will be passed in argument along with value

### CODE REVIEW

**Comment by rmdort** [comment]
Do you know why this test fails. I didnt make any change to it though

````
1 failing
  1) Date and Time Functions
       DATEVALUE:
     AssertionError: DATEVALUE("12/3") should equal 44168
: expected 44533 to be close to 44168 +/- 1e-8
````

**Comment by LesterLyu** [comment]
You can ignore this test, the test runs into different result for different year (2020 -> 2021).

**Comment by LesterLyu** [comment]
I'll merge this to v2 branch.

File: formulas/helpers.js [hunk 1]
```diff
@@ -270,6 +270,10 @@ class FormulaHelpers {
         return param.ref && !param.ref.from;
     }
 
+    retrievePosition(param) {
```
Comment (LesterLyu) [inline formulas/helpers.js:273]: This method is not necessary.

File: grammar/hooks.js [hunk 2]
```diff
@@ -152,7 +152,8 @@ class FormulaParser {
                     value: res.val,
                     isArray: res.isArray,
                     isRangeRef: !!FormulaHelpers.isRangeRef(arg),
-                    isCellRef: !!FormulaHelpers.isCellRef(arg)
+                    isCellRef: !!FormulaHelpers.isCellRef(arg),
+                    position: FormulaHelpers.retrievePosition(arg)
```
Comment (LesterLyu) [inline grammar/hooks.js:156]: Instead of calling it `position`, I will rename it to `ref` (reference position)

**Review by LesterLyu — APPROVED** [review]

**Comment by LesterLyu** [comment]
Merged to v2 branch.

**Pull request closed without merging**